tests_metricflow/snapshots/test_predicate_pushdown_rendering.py/SqlQueryPlan/DuckDB/test_fill_nulls_time_spine_metric_predicate_pushdown__plan0_optimized.sql

```diff
@@ -0,0 +1,140 @@
+          ) subq_35
+          WHERE booking__is_instant
+        ) subq_37
+        LEFT OUTER JOIN
+          ***************************.dim_listings_latest listings_latest_src_28000
+        ON
+          subq_37.listing = listings_latest_src_28000.listing_id
+        GROUP BY
+          subq_37.metric_time__day
+          , listings_latest_src_28000.country
+      ) subq_44
+      ON
+        subq_46.ds = subq_44.metric_time__day
+    ) subq_47
+  ) subq_48
+  FULL OUTER JOIN (
+    -- Compute Metrics via Expressions
+    SELECT
+      metric_time__day
+      , listing__country_latest
+      , COALESCE(bookings, 0) AS bookings_2_weeks_ago
+    FROM (
+      -- Join to Time Spine Dataset
+      SELECT
+        subq_64.ds AS metric_time__day
+        , subq_62.listing__country_latest AS listing__country_latest
+        , subq_62.bookings AS bookings
+      FROM ***************************.mf_time_spine subq_64
+      LEFT OUTER JOIN (
+        -- Constrain Output with WHERE
+        -- Pass Only Elements: ['bookings', 'listing__country_latest', 'metric_time__day']
+        -- Aggregate Measures
+        SELECT
+          metric_time__day
+          , listing__country_latest
+          , SUM(bookings) AS bookings
+        FROM (
+          -- Join Standard Outputs
+          -- Pass Only Elements: ['bookings', 'listing__country_latest', 'booking__is_instant', 'metric_time__day']
+          SELECT
+            subq_54.metric_time__day AS metric_time__day
+            , subq_54.booking__is_instant AS booking__is_instant
+            , listings_latest_src_28000.country AS listing__country_latest
+            , subq_54.bookings AS bookings
+          FROM (
+            -- Join to Time Spine Dataset
+            -- Pass Only Elements: ['bookings', 'booking__is_instant', 'metric_time__day', 'listing']
+            SELECT
+              subq_52.ds AS metric_time__day
+              , subq_50.listing AS listing
+              , subq_50.booking__is_instant AS booking__is_instant
+              , subq_50.bookings AS bookings
+            FROM ***************************.mf_time_spine subq_52
+            INNER JOIN (
+              -- Read Elements From Semantic Model 'bookings_source'
+              -- Metric Time Dimension 'ds'
+              SELECT
+                DATE_TRUNC('day', ds) AS metric_time__day
+                , listing_id AS listing
+                , is_instant AS booking__is_instant
+                , 1 AS bookings
+              FROM ***************************.fct_bookings bookings_source_src_28000
+            ) subq_50
+            ON
+              subq_52.ds - INTERVAL 14 day = subq_50.metric_time__day
+          ) subq_54
+          LEFT OUTER JOIN
+            ***************************.dim_listings_latest listings_latest_src_28000
+          ON
+            subq_54.listing = listings_latest_src_28000.listing_id
+        ) subq_59
+        WHERE booking__is_instant
```

Comment:
Is it expected that this doesn't get pushed down further? (i.e., into the subquery that directly selects from `bookings_source_src_28000`)

Reply:
Ah yes, because there's an offset in here. This one looks like it can be pushed down. I'm pretty sure I can relax that restriction now.